File: fe/src/submodules/tracker/types/initiative-project.ts
export interface InitiativeProject {
  id: string
  product_id: string
  key: string
  name: string
  description: string | null
  status: string
  owner_id: string | null
  created_at: string
  updated_at: string
  deleted_at: string | null
}

export interface CreateInitiativeProjectPayload {
  product_id: string
  key: string
  name: string
  description?: string | null
  status?: string
  owner_id?: string | null
}

export interface UpdateInitiativeProjectPayload {
  key?: string
  name?: string
  description?: string | null
  status?: string
  owner_id?: string | null
  deleted_at?: string | null
}

File: be/app/submodules/tracker/models/initiative_project.py
"""InitiativeProject model."""

from __future__ import annotations

from datetime import UTC, datetime

from beanie import DocumentWithSoftDelete, PydanticObjectId
from pydantic import Field


def _utcnow() -> datetime:
    return datetime.now(UTC)


class InitiativeProject(DocumentWithSoftDelete):
    product_id: PydanticObjectId
    key: str
    name: str
    description: str | None = None
    status: str = "planned"
    owner_id: PydanticObjectId | None = None
    created_at: datetime = Field(default_factory=_utcnow)
    updated_at: datetime = Field(default_factory=_utcnow)

    class Settings:
        name = "tracker_initiative_projects"
        indexes = ["product_id", "key", "status", "owner_id"]

    async def touch(self) -> None:
        self.updated_at = _utcnow()
        await self.save()
